File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/ReadJiraIssueActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;
using Awa.Client.Models;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Read a Jira issue using the provided request parameters.
        /// </summary>
        /// <param name="request">JiraIssueRequest containing the issue details to read.</param>
        /// <returns>The response containing the Jira issue information.</returns>
        public static async Task<JiraIssueResponse?> ReadJiraIssueActivity(JiraIssueRequest request)
        {
            return await Temporalio.Workflows.Workflow.ExecuteActivityAsync<JiraIssueResponse?>(
                AwaConstants.ActivityReadJiraIssue,
                new object?[] { request },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.McpTimeoutSeconds)
                });
        }
    }
}

File: awa/awa/sdk/utils/activity/read_jira_issue_activity.py
from datetime import timedelta

from temporalio import workflow

from awa.sdk import constants
from awa.sdk.models.jira_issue_request import JiraIssueRequest
from awa.sdk.models.jira_issue_response import JiraIssueResponse


async def read_jira_issue_activity(request: JiraIssueRequest) -> JiraIssueResponse:
    """Read a Jira issue using the provided request parameters.

    Args:
        request: JiraIssueRequest containing the issue details to read.

    Returns:
        JiraIssueResponse: The response containing the Jira issue information.

    """
    return JiraIssueResponse.model_validate(
        await workflow.execute_activity(
            constants.ACTIVITY_READ_JIRA_ISSUE,
            arg=request,
            task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
            start_to_close_timeout=timedelta(seconds=constants.MCP_TIMEOUT_SECONDS),
        ),
    )
